README.md

```diff
@@ -216,17 +216,17 @@ Notes on initializing
 
 ## Initialization is deferred by default
 
-Konfa will initialize its values when a variable is first read (when for instance
-calling `get`, `true?` or `dump`). Initialization can be triggered earlier by calling
-the `init`method.
+Konfa will initialize its values as late as possible, in other words, when a
+variable is first read (when for instance calling `get`, `true?` or `dump`).
+However, it's recommended to trigger the initialization as early as possible,
+as it will then fail early if bad configuration values are found.
```

Comment:
Possible silly question: 

Now that we support initialisation with multiple files in one call to `read_from` or `init_with`  do we want to continue supporting the lazy initialization use-case? 

I understand that this is open source and we want to avoid needless breaking changes, however it read a little odd that we describe the default behaviour in one sentence, and then advise against it in the next.

Let me know what you think :)

Reply:
That's a very good point and tbh I was questioned the same thing when writing. I don't think I can answer that though because I'm not fully aware of the use-cases for lazy loading either. Any thoughts, @ghansson?

Reply:
The behaviour stems from the original requirement of loading configuration from different sources and different initialisors and the idea that the consuming code gets to set up configuration initalisation after its requirements, leaving the library to provide functionality only.

The changes to `read_from` breaks this as it's forcing your to choose initialisor once, then read from whatever source/file/database or thing.

The introduction of `read_from` makes more assumptions about the environment and context in which the consuming code than the original implementation. I'm not sure I like that in a general purpose library like this.

Reply:
> The behaviour stems from the original requirement of loading configuration from different sources and different initialisors and the idea that the consuming code gets to set up configuration initalisation after its requirements, leaving the library to provide functionality only.

Thanks for the historic background but I think the question remains: why does the documentation advise against the default behavior? This PR just rephrases that part but doesn't change the message itself.

> The changes to read_from breaks this as it's forcing your to choose initialisor once, then read from whatever source/file/database or thing.

> The introduction of read_from makes more assumptions about the environment and context in which the consuming code than the original implementation. I'm not sure I like that in a general purpose library like this.

What do you mean? The new interface was designed to match the old behavior while making it more straightforward and improving how one loads multiple configurations; it wasn't complete until now though (with #20 and this PR).

However, I share your concern about the overall change (going back from #15 to this) but since we had long started it I wanted to have the full implementation first. In hindsight, I think the improvements to the interface and the code under the hood could have been done in a different way. I'd be keen to taking a step back and discussing again how to better achieve the goal of the new interface in another way. Should we go ahead with the PR and then take it from there?